### PR TITLE
Ddcnls 3833:Add regex validation and update tests

### PR DESCRIPTION
**Pull request by ashwinirudrappa**

### CODE REVIEW

File: app/forms/AtedForms.scala [hunk 1]
```diff
@@ -41,48 +43,55 @@ object AtedForms {
...
+  val telephoneRegex = "^[A-Z0-9 )/(\\-*#]+$".r
   val emailRegex =
     """^(?!\.)("([^"\r\\]|\\["\r\\])*"|([-a-zA-Z0-9!#$%&'*+/=?^_`{|}~]|(?<!\.)\.)*)
       |(?<!\.)@[a-zA-Z0-9][\w\.-]*[a-zA-Z0-9]\.[a-zA-Z][a-zA-Z\.]*[a-zA-Z]$""".r
-  val postCodeRegex = """^[A-Z]{1,2}[0-9R][0-9A-Z]? [0-9][ABD-HJLNP-UW-Z]{2}$""".r
-  val addressLineRegex = """^([A-Za-z0-9\s\&\-\,\@\#\.\'])*$""".r
+  val postCodeRegex = "^[A-Z]{1,2}[0-9][0-9A-Z]?\\s?[0-9][A-Z]{2}|BFPO\\s?[0-9]{1,10}$".r
+  val addressLineRegex = "^[A-Za-z0-9 \\-,.&']{1,35}$".r
+  val addressRegex = "^[A-Za-z0-9 \\-,.&']{1,35}$"
+  val nameRegex = "^[a-zA-Z &`\\-\'^]{1,35}$"
```
Comment (Gaur4vGaur) [inline app/forms/AtedForms.scala]: Please follow best practices to name constants. Would it be worth moving these to constant file?

File: app/forms/BankDetailForms.scala [hunk 2]
```diff
@@ -30,6 +30,9 @@ import scala.util.Try
 object BankDetailForms {
 
   val TWO = 2
+  val accountNameRegex = "^[a-zA-Z][a-zA-Z '.& \\/]{1,59}$"
+  val swiftCodeRegex = "^[A-Za-z0-9 \\-]{1,11}$"
+  val ibanRegex = "^[A-Za-z0-9 \\-]{1,34}$"
```
Comment (Gaur4vGaur) [inline app/forms/BankDetailForms.scala]: Please follow best practices to name constants. Would it be worth moving these to constant file?

File: app/forms/OverseasCompanyRegistrationForm.scala [hunk 3]
```diff
@@ -22,19 +22,31 @@ import play.api.data.Forms._
...
+import uk.gov.hmrc.play.mappers.StopOnFirstFail
+import uk.gov.hmrc.play.mappers.StopOnFirstFail.constraint
 
 object OverseasCompanyRegistrationForm {
 
   private val length60 = 60
   private val length40 = 40
+  private val idNumberRegex = "^[a-zA-Z0-9 '&\\-]{1,60}$"
+  private val issuingInstitutionRegex = "^[a-zA-Z0-9 '&\\-\\/]{1,40}$"
+  private val issuingCountryRegex = "(?!^GB$)^[A-Z]{2}$"
```
Comment (Gaur4vGaur) [inline app/forms/OverseasCompanyRegistrationForm.scala]: Please follow best practices to name constants. Would it be worth moving these to constant file?

File: app/forms/PropertyDetailsForms.scala [hunk 4]
```diff
@@ -51,41 +56,45 @@ object PropertyDetailsForms {
     """^(?!\.)("([^"\r\\]|\\["\r\\])*"|([-a-zA-Z0-9!#$%&'*+/=?^_`{|}~]|(?<!\.)\.)*)
       |(?<!\.)@[a-zA-Z0-9][\w\.-]*[a-zA-Z0-9]\.[a-zA-Z][a-zA-Z\.]*[a-zA-Z]$""".r
   val supportingInfoRegex = """^[A-Za-z0-9\s\.\,\']*$"""
-  val titleRegex = """^[A-Za-z0-9\s\.\,\']*$"""
+  val titleRegex = "^[A-Za-z0-9 ,.&']{1,40}$"
```
Comment (Gaur4vGaur) [inline app/forms/PropertyDetailsForms.scala]: Please follow best practices to name constants. Would it be worth moving these to constant file?

File: test/forms/AtedFormsSpec.scala [hunk 5]
```diff
@@ -32,19 +32,61 @@ class AtedFormsSpec extends PlaySpec with OneServerPerSuite {
...
+
+      "fail form validation when form data exceeds max length" in {
+        AtedForms.editContactDetailsForm.bind(maxLengthFormData).fold(
+          formWithErrors => {
+            formWithErrors.errors.head.message must be("First name cannot be more than 35 characters")
+            formWithErrors.errors(1).message must be("Last name cannot be more than 35 characters")
+            formWithErrors.errors(2).message must be("Telephone number must not be more than 24 characters")
+            formWithErrors.errors.length must be(3)
+          },
+          success => {
```
Comment (Gaur4vGaur) [inline test/forms/AtedFormsSpec.scala]: would it be worth calling fail if binding succeed?

File: test/forms/AtedFormsSpec.scala [hunk 6]
```diff
@@ -32,19 +32,61 @@ class AtedFormsSpec extends PlaySpec with OneServerPerSuite {
...
+
+     "fail form validation on invalid form data" in {
+       AtedForms.editContactDetailsForm.bind(invalidFormData).fold(
+         formWithErrors => {
+           formWithErrors.errors.head.message must be("ated.contact-details-first-name.invalid")
+           formWithErrors.errors(1).message must be("ated.contact-details-first-name.invalid")
+           formWithErrors.errors(2).message must be("Telephone number is not valid")
+           formWithErrors.errors.length must be(3)
+         },
+         success => {
```
Comment (Gaur4vGaur) [inline test/forms/AtedFormsSpec.scala]: would it be worth calling fail if binding succeed?

File: test/views/editLiability/bankDetailsSpec.scala [hunk 7]
```diff
@@ -26,7 +26,8 @@ import org.scalatest.{BeforeAndAfterEach, FeatureSpec, GivenWhenThen}
 import org.scalatestplus.play.OneServerPerSuite
 import play.api.test.FakeRequest
 
-class bankDetailsSpec extends FeatureSpec with OneServerPerSuite with MockitoSugar with BeforeAndAfterEach with GivenWhenThen{
+class
+bankDetailsSpec extends FeatureSpec with OneServerPerSuite with MockitoSugar with BeforeAndAfterEach with GivenWhenThen{
```
Comment (Gaur4vGaur) [inline test/views/editLiability/bankDetailsSpec.scala]: this degrades readability. I would refrain from refactoring it

File: test/forms/AtedFormsSpec.scala (hunk 5 above)
Comment (ashwinirudrappa) [reply test/forms/AtedFormsSpec.scala]: Fixed it

File: app/forms/AtedForms.scala (hunk 1 above)
Comment (ashwinirudrappa) [reply app/forms/AtedForms.scala]: Fixed it

File: app/forms/BankDetailForms.scala (hunk 2 above)
Comment (ashwinirudrappa) [reply app/forms/BankDetailForms.scala]: Fixed it

File: app/forms/PropertyDetailsForms.scala (hunk 4 above)
Comment (ashwinirudrappa) [reply app/forms/PropertyDetailsForms.scala]: Done the code changes

File: test/views/editLiability/bankDetailsSpec.scala (hunk 7 above)
Comment (ashwinirudrappa) [reply test/views/editLiability/bankDetailsSpec.scala]: fixed

**Pull request closed without merging**